src/fs.rs

```diff
@@ -512,7 +552,7 @@ impl<IO: Read + Write + Seek, TP, OCC> FileSystem<IO, TP, OCC> {
     ///
     /// `Error::Io` will be returned if the underlying storage object returned an I/O error.
     pub fn read_status_flags(&self) -> Result<FsStatusFlags, Error<IO::Error>> {
-        let bpb_status = self.bpb.status_flags();
+        let bpb_status = self.current_status_flags.get();
```

Comment:
this function allowed to check if dirty flag was set before filesystem was opened. `current_status_flags` will never have this flag set, even if `ignore_dirty_flag` is set. I would rather keep the old behavior for now because new one makes it impossible to mount a dirty volume and still get information about it being dirty.